web/src/components/IdeViewer/IdeViewer.js

```diff
@@ -124,8 +124,7 @@ const IdeViewer = () => {
   useEffect(() => {
     setImage(
       state.objectData?.type === 'png' &&
-        state.objectData?.data &&
-        window.URL.createObjectURL(state.objectData?.data)
+        'data:image/png;base64,' + state.objectData?.data
```

Comment:
The base64 image can just about be thrown directly into `src`, makes this a little simpler.